src/medals/main.cljs

```diff
@@ -21,22 +21,19 @@
                   [:span code]]]
                 (map (fn [key] ^{:key key} [:td (get country key)]) (filter (fn [key] (contains? ALLOWED_SORT_VALUES key)) (keys country)))])
 
-(defn get-countries-with-total [countries]
-  (map (fn [country] (assoc country :total (get-total-count country))) countries))
-
 
 (defn render-widget []
   (let [sort @(re-frame/subscribe [::subs/sort])
-        countries (get-countries-with-total @(re-frame/subscribe [::subs/countries]))]
+        countries (sort-countries (get-countries-with-total @(re-frame/subscribe [::subs/countries])) sort)]
```

Comment:
All this logic should live in subscriptions. This will make it easily reusable in multiple views and have performance benefits because subscriptions memoize their results based on the current state of the DB.

Make a `countries-with-total` subscription which depends on the `countries` sub  and have it run `get-countries-with-total` then make a `sorted-countries` sub that depends on the `countries-with-total` sub.